core/src/main/scala/com/salesforce/op/stages/impl/insights/RecordInsightsLOCO.scala

```diff
@@ -143,17 +148,18 @@ class RecordInsightsLOCO[T <: Model[T]]
 
       // Let's check the indicator value and descriptor value
       // If those values are empty, the field is likely to be a derived text feature (hashing tf output)
-      if (history.indicatorValue.isEmpty && history.descriptorValue.isEmpty) {
+      if (textFeatureIndices.contains(oldInd) && history.indicatorValue.isEmpty && history.descriptorValue.isEmpty) {
         // Name of the field
-        val rawName = history.parentFeatureStages match {
-          case s if s.exists(_.contains(smartTextClassName)) => history.parentFeatureOrigins.headOption
-          case s if s.exists(_.contains(smartTextMapClassName)) => history.grouping
-          case _ => None
+        val rawName = history.parentFeatureType match {
+          case s if s.exists(mapTypes.contains) => history.grouping
+          case s if s.exists(textTypes.contains) => history.parentFeatureOrigins.headOption
+          case s => throw new Error(s"type should be Text or TextMap, here ${s.mkString(",")}")
         }
         // Update the aggregation map
         for {name <- rawName} {
-          val (indices, array) = aggregationMap.getOrElse(name, (Array.empty[Int], Array.empty[Double]))
-          aggregationMap.update(name, (indices :+ i, sumArrays(array, diffToExamine)))
+          val key = name + "_" + history.parentFeatureStages.mkString(",")
```

Comment:
the key value can be quite large is do concatenate all the parent stages. what's the rationale behind it? @mweilsalesforce

Reply:
It is possible to apply different transformations on a same feature.
After thought, maybe we should aggregate all the derived features even if 2 different transformations were applied

Reply:
Ok fixed